docs/source/entities.mdx

```diff
@@ -0,0 +1,321 @@
+
+**Every instance of an entity must be uniquely identifiable by its `@key` fields.** This is what enables your gateway to associate data from _different_ subgraphs with the _same_ entity instance.
+
+> An entity's `@key` cannot include fields that return a union or interface.
+
+## Referencing an entity
+
+After you define an entity in one subgraph, _other_ subgraphs can then reference that entity in their schema.
+
+Recall our example `Product` entity from the Products subgraph:
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+With this definition in place, the Reviews subgraph can then add a field of type `Product` to its `Review` type, like so:
+
+```graphql{3}
+# Reviews subgraph
+type Review {
+  product: Product!
+  score: Int!
+}
+
+# This is a required "stub" of the Product entity (see below)
+type Product {
+  id: ID!
+}
+```
+
+To reference the `Product` entity, the Reviews subgraph needs to define a **stub** of it to make its own schema valid. The stub _doesn't_ need to include all `Product` fields. It _only_ needs to include `id`, because that's the entity's `@key` field. Without this field, the gateway doesn't know how to fetch the correct `Product` for a given `Review`.
+
+> Notice that the stub doesn't even need a `@key` directive! That's because the Reviews subgraph isn't adding any of its _own_ fields to `Product`. We'll see that later in [Extending entities](#extending-entities).
+
+This syntax has several benefits:
+
+* It's standard GraphQL grammar.
+* It enables you to run the `reviews` subgraph standalone with a valid schema, including a `Product` type with a single `id` field.
+* It provides strong typing information that lets you catch mistakes at schema composition time.
+
+## Resolving entities
+
+Let's say our Reviews subgraph from [Referencing an entity](#referencing-an-entity) defines the following `Query` type:
+
+```graphql:title=reviews
+type Query {
+  latestReviews: [Review!]
+}
+```
+
+That means the following query is valid against our federated graph:
+
+```graphql{6}
+query GetReviewsAndProducts {
+  latestReviews { # Defined in Reviews
+    score
+    product {
+      id
+      price # Not defined in Reviews!
+    }
+  }
+}
+```
+
+Now we have a problem: this query starts its execution in the Reviews subgraph (where `latestReviews` is defined), but that subgraph _doesn't_ know that `Product` entities have a `price` field! Remember, the Reviews subgraph only knows about the `id` field of `Product`.
+
+Because of this, the gateway needs to fetch `price` from the Products subgraph instead. But how does the gateway know _which_ products it needs to fetch the prices for?
+
+To solve this, we add a resolver to _each_ subgraph:
+
+* A resolver in Reviews to generate [representations](#entity-representations) of `Product` entities
```

Comment:
I'm not really sure about that one, and I'm particularly uneasy with the presentation of this alongside the "reference resolver".

I'm not sure about the concept of "a resolver to generate entity representations" in general because it's usually not 1 resolver, it's not a particularly special resolver, and I think the concept of an "entity representation" is not that clearly defined in most cases. 

What I mean here is that "entity representations" are returned by the "normal" resolvers of your subgraph. In the example above, the resolver that would return `Product` "representation" is the one of `latestReviews`, but that resolver is a normal, non-federation-specific, resolver. And What it returns is also not very federation-specific in the sense that it should return `Review` objects, and those should have a `product` field that is itself an object containing an `id`, but all that is what the graphQL schema declare. By which I mean, if `Reviews` was not a federated schema at all, that `latestReviews` would be _exactly_ the same.

Also, there is usually more than on resolver in a subgraph returning entities representations. Typically, if you had a 2nd operation, say `Query.bestReviews`, then the resolver of that would also return some `Product` "representations". But the phrasing here suggest a single resolver for all `Product` entities representations.

My personal suggestion here would be to say that there is actually nothing special to do in Reviews that is federation specific. You just need to implement resolvers as you would if Review wasn't federated in that case. And then I would say that Product does need a special resolver for its `@key`.

Reply:
I've updated this section significantly to only focus on reference resolvers, and note that the already-present resolvers for the other subgraph do the trick of returning representations 👍